lib/rubocop/cop/lint/duplicate_set_element.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for duplicate elements in Set.
+      #
+      # @example
+      #
+      #   # bad
+      #   Set[:foo, :bar, :foo]
+      #
+      #   # good
+      #   Set[:foo, :bar]
+      #
+      #   # bad
+      #   Set.new([:foo, :bar, :foo])
+      #
+      #   # good
+      #   Set.new([:foo, :bar])
+      #
+      class DuplicateSetElement < Base
+        extend AutoCorrector
+
+        MSG = 'Remove the duplicate element in Set.'
+        RESTRICT_ON_SEND = %i[\[\] new].freeze
+
+        # @!method set_init_elements(node)
+        def_node_matcher :set_init_elements, <<~PATTERN
+          {
+            (send (const {nil? cbase} :Set) :[] $...)
+            (send (const {nil? cbase} :Set) :new (array $...))
+          }
+        PATTERN
+
+        def on_send(node)
+          return unless (set_elements = set_init_elements(node))
+
+          seen_elements = Set[]
+
+          set_elements.each_with_index do |set_element, index|
+            # NOTE: Ignore due to the possibility of the corner case where Set elements
```

Comment:
checking for `call_type?` is probably not enough, since this would still wrongly flag and correct cases where the method call is buried in some expression, like:
```ruby
Set[(1 + rand), (1 + rand)]
Set[[rand], [rand]]
Set[rand > 0.5 ? 1 : 2, rand > 0.5 ? 1 : 2]
```

it might be safest to just restrict this cop to literals, constants, and variables than try to detect every edge case

Reply:
Ah, this was a point I was also considering. I've updated this PR. Thank you!